### PR TITLE
Fix list feature format for BigQuery offline datasources. 

### LINKED ISSUE #1839: Array/list feature materialization in BQ crashes in type conversion
Currently, if you try to use BQ and materialize a feature that is a list (of numbers, strings, etc), Feast will crash because in BQ, the value type of the feature is a dictionary, such as 

`{'list': [{'item': 3}, {'item': 3}]}`
In materialize, we convert the latest values retrieval job to a pyarrow table and then converts to ValueProtos to write. This calls 

`python_type_to_feast_value_type`, which doesn't support dict types.

### PR DESCRIPTION
**Pull request by judahrand**
**What this PR does / why we need it**:
Should have been using this option from the start but the docs on it are very lacking. Took me an hour or two to find it. This should make BQ behave like the other data store with `pyarrow` lists instead of the weird `{list: {item: value}}`.

Some concerns about this being incompatible with already existing tables? Though I think really this is important as the transformation after the fact is slow and this it the 'right' fix.

**Which issue(s) this PR fixes**:

Fixes #1839

**Does this PR introduce a user-facing change?**:

```release-note
Fix list feature format for BigQuery offline datasources. 
```

### CODE REVIEW

**Comment by judahrand** [comment]
Not quite there 😢

File: sdk/python/tests/integration/registration/test_universal_types.py [hunk 1]
```diff
@@ -255,16 +248,10 @@ def assert_feature_list_types(
...
-        assert (
-            feature_list_dtype_to_expected_historical_feature_list_dtype[feature_dtype]
-            in type(historical_features_df.value[0]["list"][0]["item"]).__name__
-        )
-    else:
-        assert (
-            feature_list_dtype_to_expected_historical_feature_list_dtype[feature_dtype]
-            in type(historical_features_df.value[0][0]).__name__
-        )
+    assert (
```
Comment (adchia) [inline sdk/python/tests/integration/registration/test_universal_types.py:251]: think you'll need to revert this change that you did on the left

**Comment by judahrand** [comment]
All tests passing.

**Review by adchia — APPROVED** [review]
/lgtm
